cmd/ctr/pprof.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"github.com/urfave/cli"
+)
+
+type pprofDialer struct {
+	proto string
+	addr  string
+}
+
+var pprofCommand = cli.Command{
+	Name:  "pprof",
+	Usage: "provides golang pprof outputs for containerd",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "debug-socket, d",
+			Usage: "socket path for containerd's debug server",
+			Value: "/run/containerd/containerd-debug.sock",
+		},
+	},
+	Subcommands: []cli.Command{
+		pprofGoroutinesCommand,
+		pprofHeapCommand,
+		pprofProfileCommand,
+		pprofTraceCommand,
+		pprofBlockCommand,
+		pprofThreadcreateCommand,
+	},
+}
+
+var pprofGoroutinesCommand = cli.Command{
+	Name:  "goroutines",
+	Usage: "dump goroutine stack dump",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "full,f",
+			Usage: "full goroutine stack dump",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		var uri string
+		client := getPProfClient(context)
+
+		full := context.Bool("full")
```

Comment:
There are very few cases where we wouldn't want full debug output. Let's just use `debug=2` as the default and remove the flag.